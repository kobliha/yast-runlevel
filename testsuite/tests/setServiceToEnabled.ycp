// test for serServiceToEnabled
{
    include "testsuite.ycp";
    include "runlevel/ui.ycp";

    map start_variables_disabled = $[];
    map services = $[];

    // has not START_ variable
    start_variables_disabled = $[];
    services = $[
	"a" : $[
	    "start" : [ "1", "2" ],
	    "defstart" : [ "3", "4" ],
	    ]
	];
    setServiceToEnabled ("a");
    DUMP (services);

    start_variables_disabled = $[];
    services = $[
	"a" : $[
	    "START" : true,
	    "start" : [ "1", "2" ],
	    "defstart" : [ "3", "4" ],
	    ]
	];
    setServiceToEnabled ("a");
    DUMP (services);

    start_variables_disabled = $[];
    services = $[
	"a" : $[
	    "START" : false,
	    "start" : [ "1", "2" ],
	    "defstart" : [ "3", "4" ],
	    ]
	];
    setServiceToEnabled ("a");
    DUMP (services);

    // has START_ variable
    start_variables_disabled = $["a" : "START_a"];
    services = $[
	"a" : $[
	    "start" : [ "1", "2" ],
	    "defstart" : [ "3", "4" ],
	    ]
	];
    setServiceToEnabled ("a");
    DUMP (services);

    start_variables_disabled = $["a" : "START_a"];
    services = $[
	"a" : $[
	    "START" : true,
	    "start" : [ "1", "2" ],
	    "defstart" : [ "3", "4" ],
	    ]
	];
    setServiceToEnabled ("a");
    DUMP (services);

    start_variables_disabled = $["a" : "START_a"];
    services = $[
	"a" : $[
	    "START" : false,
	    "start" : [ "1", "2" ],
	    "defstart" : [ "3", "4" ],
	    ]
	];
    setServiceToEnabled ("a");
    DUMP (services);


}
