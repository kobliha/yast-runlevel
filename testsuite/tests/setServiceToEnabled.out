Dump	$["a":$["START":true, "defstart":["3", "4"], "dirty":true, "start":["3", "4"]]]
Dump	$["a":$["START":true, "defstart":["3", "4"], "dirty":true, "start":["3", "4"]]]
Dump	$["a":$["START":true, "defstart":["3", "4"], "dirty":true, "start":["3", "4"]]]
Dump	$["a":$["START":true, "defstart":["3", "4"], "dirty":true, "start":["3", "4"]]]
Dump	$["a":$["START":true, "defstart":["3", "4"], "dirty":true, "start":["3", "4"]]]
Dump	$["a":$["START":true, "defstart":["3", "4"], "dirty":true, "start":["3", "4"]]]
